app/uk/gov/hmrc/exports/services/reversemapping/declaration/items/SingleItemParser.scala

```diff
@@ -17,19 +17,45 @@
 package uk.gov.hmrc.exports.services.reversemapping.declaration.items
 
 import java.util.UUID
+
 import scala.xml.NodeSeq
+
 import javax.inject.Inject
-import uk.gov.hmrc.exports.models.declaration.ExportItem
+import uk.gov.hmrc.exports.models.declaration.YesNoAnswer.YesNoAnswers
+import uk.gov.hmrc.exports.models.declaration._
+import uk.gov.hmrc.exports.services.reversemapping.MappingContext
 import uk.gov.hmrc.exports.services.reversemapping.declaration.DeclarationXmlParser
-import uk.gov.hmrc.exports.services.reversemapping.declaration.DeclarationXmlParser.XmlParserResult
+import uk.gov.hmrc.exports.services.reversemapping.declaration.DeclarationXmlParser._
 import uk.gov.hmrc.exports.services.reversemapping.declaration.XmlTags._
-import uk.gov.hmrc.exports.services.reversemapping.MappingContext
 
 class SingleItemParser @Inject()(procedureCodesParser: ProcedureCodesParser) extends DeclarationXmlParser[ExportItem] {
 
   override def parse(itemXml: NodeSeq)(implicit context: MappingContext): XmlParserResult[ExportItem] =
-    procedureCodesParser.parse(itemXml).map {
-      val sequenceId = (itemXml \ SequenceNumeric).text.toInt
-      ExportItem(UUID.randomUUID().toString, sequenceId, _)
+    procedureCodesParser.parse(itemXml).map { procedureCodes =>
+      val domesticDutyTaxParties = itemXml \ DomesticDutyTaxParty
+
+      ExportItem(
+        id = UUID.randomUUID().toString,
+        sequenceId = (itemXml \ SequenceNumeric).text.toInt,
+        procedureCodes = procedureCodes,
+        fiscalInformation = parseFiscalInformation(domesticDutyTaxParties),
+        additionalFiscalReferencesData = parseAdditionalFiscalReferencesData(domesticDutyTaxParties),
+        statisticalValue = parseStatisticalValue(itemXml)
+      )
     }
+
+  private def parseAdditionalFiscalReferencesData(domesticDutyTaxParties: NodeSeq): Option[AdditionalFiscalReferences] = {
```

Comment:
IMHO this part deserves its own parser. It makes sense not to create separate parsers for very simple cases, but here you have more logic inside and therefore the test suite is also bigger.

Reply:
My 2 cents... Personally I do not see the advantage to have a separate parser for this simple logic (I have seen other more complicated "internal methods" than this one :wink:). And more in general I'm against to tailor the implementation according to the tests. I would rather have an additional test class if required. However, always open to have a discussion of course.

Reply:
I'm happy for this to be a private method inside this parser. :+1: